### PR TITLE
Code Quality Fixes(10)

### PR DESCRIPTION
**Pull request by lakshyacis**
bitmessagemain quality fixes
bmconfigparser quality fixes
class_addressGenerator quality fixes
class_objectProcessor quality fixes
class_singleCleaner.py quality fixes
class_singleWorker quality fixes
class_smtpServer quality fixes

### CODE REVIEW

**Comment by PeterSurda** [comment]
Well as you see, travis breaks.

File: src/bitmessagemain.py [hunk 1]
```diff
@@ -68,6 +72,7 @@
 
 
 def connectToStream(streamNumber):
+    """This function helps us to connect with the stream"""
```
Comment (PeterSurda) [inline src/bitmessagemain.py]: `"""Connect to a stream"""`

File: src/bitmessagemain.py [hunk 2]
```diff
@@ -174,16 +181,17 @@ def signal_handler(signum, frame):
     if shared.thisapp.daemon or not state.enableGUI:
         shutdown.doCleanShutdown()
     else:
-        print('# Thread: %s(%d)' % (thread.name, thread.ident))
+        print '# Thread: %s(%d)' % (thread.name, thread.ident)
```
Comment (PeterSurda) [inline src/bitmessagemain.py]: I disagree, this is for python3 compatibility. Probably better to suppress the warning.

File: src/bitmessagemain.py [hunk 3]
```diff
@@ -174,16 +181,17 @@ def signal_handler(signum, frame):
...
-        print('Unfortunately you cannot use Ctrl+C when running the UI'
-              ' because the UI captures the signal.')
+                print'  %s' % line.strip()
+        print 'Unfortunately you cannot use Ctrl+C when running the UI \
+        because the UI captures the signal.'
 
 
-class Main:
+class Main:    # pylint: disable=no-init, old-style-class, no-self-use
+    """Module starts the proxy config plugin"""
```
Comment (PeterSurda) [inline src/bitmessagemain.py]: `"""Main PyBitmessage class"""`

File: src/bitmessagemain.py [hunk 4]
```diff
@@ -206,22 +214,23 @@ def start_proxyconfig(config):
                     'Started proxy config plugin %s in %s sec',
                     proxy_type, time.time() - proxyconfig_start)
 
-    def start(self):
+    def start(self):    # pylint: disable=too-many-statements, too-many-branches, too-many-locals
+        """Starting the daemon"""
```
Comment (PeterSurda) [inline src/bitmessagemain.py]: `"""Start main application"""`

File: src/bitmessagemain.py [hunk 5]
```diff
@@ -404,7 +413,7 @@ def start(self):
             if state.curses:
                 if not depends.check_curses():
                     sys.exit()
-                print('Running with curses')
+                print 'Running with curses'
```
Comment (PeterSurda) [inline src/bitmessagemain.py]: Suppress, py3 compatibility

File: src/bitmessagemain.py [hunk 6]
```diff
@@ -505,12 +517,14 @@ def usage(self):
 '''
 
     def stop(self):
+        """Stopping Bitmessage Deamon"""
```
Comment (PeterSurda) [inline src/bitmessagemain.py]: `"""Stop main application"""`

File: src/bitmessagemain.py [hunk 7]
```diff
@@ -505,12 +517,14 @@ def usage(self):
...
         with shared.printLock:
-            print('Stopping Bitmessage Deamon.')
+            print 'Stopping Bitmessage Deamon.'
```
Comment (PeterSurda) [inline src/bitmessagemain.py]: py3 compatibility

File: src/class_objectProcessor.py [hunk 8]
```diff
@@ -268,6 +275,7 @@ def processgetpubkey(data):
             queues.workerQueue.put(('sendOutOrStoreMyV4Pubkey', myAddress))
 
     def processpubkey(self, data):
+        """Pubkey processed"""
```
Comment (PeterSurda) [inline src/class_objectProcessor.py]: `"""Process a pubkey object"""`

File: src/class_objectProcessor.py [hunk 9]
```diff
@@ -438,6 +446,7 @@ def processpubkey(self, data):
             timeRequiredToProcessPubkey)
 
     def processmsg(self, data):
+        """"""
```
Comment (PeterSurda) [inline src/class_objectProcessor.py]: `"""Process a message object"""`

File: src/class_objectProcessor.py [hunk 10]
```diff
@@ -784,6 +793,7 @@ def processmsg(self, data):
         )
 
     def processbroadcast(self, data):
+        """"""
```
Comment (PeterSurda) [inline src/class_objectProcessor.py]: `"""Process a broadcast object"""`

File: src/class_objectProcessorQueue.py [hunk 11]
```diff
@@ -1,23 +1,32 @@
...
+
 class ObjectProcessorQueue(Queue.Queue):
+    """A queue for processsing message objects"""
     maxSize = 32000000
 
     def __init__(self):
         Queue.Queue.__init__(self)
         self.sizeLock = threading.Lock()
-        self.curSize = 0 # in Bytes. We maintain this to prevent nodes from flooing us with objects which take up too much memory. If this gets too big we'll sleep before asking for further objects.
+        self.curSize = 0
```
Comment (PeterSurda) [inline src/class_objectProcessorQueue.py]: Line swap.

File: src/class_singleCleaner.py [hunk 12]
```diff
@@ -186,6 +189,7 @@ def resendPubkeyRequest(address):
 
 
 def resendMsg(ackdata):
+    """Resending acknowledgement"""
```
Comment (PeterSurda) [inline src/class_singleCleaner.py]: `"""Resend message (acknowledgement expired without being received)"""`

File: src/bitmessagemain.py (hunk 2 above)
Comment (lakshyacis) [reply src/bitmessagemain.py]: Okay

**Comment by lakshyacis** [comment]
> Please change docstrings and the comments as explained above.

Done.

File: src/bitmessagemain.py [hunk 13]
```diff
@@ -183,7 +190,8 @@ def signal_handler(signum, frame):
               ' because the UI captures the signal.')
 
 
-class Main:
+class Main(object):
```
Comment (lakshyacis) [inline src/bitmessagemain.py]: @g1itch @PeterSurda 
I tested and work fine.
Is it okay to use object here?

Comment (g1itch) [reply src/bitmessagemain.py]: Should be OK. I don't understand the reason of using old classes at all. Maybe it was a sort of optimization. If you don't inherit from both old and new style classes it's the right way to use new style classes in python 2.7.

Comment (PeterSurda) [reply src/bitmessagemain.py]: I also think it's ok.

File: src/class_objectProcessor.py [hunk 14]
```diff
@@ -1,10 +1,14 @@
...
+=================================
+"""
 import hashlib
 import random
-import shared
 import threading
 import time
 from binascii import hexlify
 from subprocess import call  # nosec
+import shared
```
Comment (g1itch) [inline src/class_objectProcessor.py]: why?

File: src/class_addressGenerator.py [hunk 15]
```diff
@@ -135,8 +140,8 @@ def run(self):
                         potentialPubSigningKey + potentialPubEncryptionKey)
                     ripe = RIPEMD160Hash(sha.digest()).digest()
                     if (
-                        ripe[:numberOfNullBytesDemandedOnFrontOfRipeHash] ==
-                        '\x00' * numberOfNullBytesDemandedOnFrontOfRipeHash
+                            ripe[:numberOfNullBytesDemandedOnFrontOfRipeHash] ==
+                            '\x00' * numberOfNullBytesDemandedOnFrontOfRipeHash
```
Comment (g1itch) [inline src/class_addressGenerator.py]: why?

File: src/bmconfigparser.py [hunk 16]
```diff
@@ -1,4 +1,7 @@
 """
+src/bmconfigparser.py
+=================================
+
```
Comment (g1itch) [inline src/bmconfigparser.py]: Please don't add this garbage

File: src/class_objectProcessor.py (hunk 14 above)
Comment (g1itch) [inline src/class_objectProcessor.py]: Please add a blank line

File: src/class_singleCleaner.py [hunk 17]
```diff
@@ -36,11 +36,12 @@
 
 
 class singleCleaner(StoppableThread):
+    """Module run singleCleaner thread"""
     name = "singleCleaner"
     cycleLength = 300
     expireDiscoveredPeers = 300
 
-    def run(self):
+    def run(self):        # pylint: disable=too-many-branches, protected-access
```
Comment (g1itch) [inline src/class_singleCleaner.py]: two spaces is enough

File: src/class_singleCleaner.py [hunk 18]
```diff
@@ -128,10 +129,10 @@ def run(self):
                              "MainWindow",
                              'Alert: Your disk or data storage volume'
                              ' is full. Bitmessage will now exit.'),
-                            True)
+                         True)
                     ))
                     # FIXME redundant?
-                    if shared.daemon or not state.enableGUI:
+                    if shared.thisapp.daemon or not state.enableGUI:
```
Comment (g1itch) [inline src/class_singleCleaner.py]: are you sure?

File: src/class_objectProcessor.py (hunk 14 above)
Comment (lakshyacis) [reply src/class_objectProcessor.py]: Pylint warning: wrong-import-order. 
Standard import " import threading, import time, from binascii import hexlify, from subprocess import call " should be placed before "import shared"

File: src/class_addressGenerator.py (hunk 15 above)
Comment (lakshyacis) [reply src/class_addressGenerator.py]: Pylint warning: Wrong hanging indentation before a block.

File: src/bmconfigparser.py (hunk 16 above)
Comment (lakshyacis) [reply src/bmconfigparser.py]: Okay, I will remember from next time.

File: src/class_objectProcessor.py (hunk 14 above)
Comment (lakshyacis) [reply src/class_objectProcessor.py]: Okay. Thanks

File: src/class_addressGenerator.py (hunk 15 above)
Comment (g1itch) [reply src/class_addressGenerator.py]: What version? So how do you think I merged it?

File: src/class_singleCleaner.py (hunk 17 above)
Comment (lakshyacis) [reply src/class_singleCleaner.py]: Okay, I will remember from next time, and I will change as requested.

File: src/class_singleCleaner.py (hunk 18 above)
Comment (lakshyacis) [reply src/class_singleCleaner.py]: I am pretty sure and haven't tested. But while working on this file. I checked the directory of shared and there is no daemon in shared. And even no daemon is found in shared.thisapp. So I took the reference from line no. 77
" if shared.thisapp.daemon or not state.enableGUI: "

Comment (lakshyacis) [reply src/class_singleCleaner.py]: I am pretty sure. But while working on this file. I checked the directory of shared and there is no daemon in shared. So I took the reference from line no. 77
" if shared.thisapp.daemon or not state.enableGUI: "

File: src/class_addressGenerator.py (hunk 15 above)
Comment (lakshyacis) [reply src/class_addressGenerator.py]: I'm currently using pylint 1.9.5. Maybe checks using an old version of pylint.

Comment (g1itch) [reply src/class_addressGenerator.py]: https://github.com/PyCQA/pylint/issues/289

File: src/class_objectProcessor.py (hunk 14 above)
Comment (g1itch) [reply src/class_objectProcessor.py]: [PEP8](https://www.python.org/dev/peps/pep-0008/#imports) requires that explicitly:

> You should put a blank line between each group of imports.

File: src/class_addressGenerator.py (hunk 15 above)
Comment (g1itch) [reply src/class_addressGenerator.py]: It applies to all such expressions. I removed extra spaces intentionally.

File: src/bmconfigparser.py (hunk 16 above)
Comment (g1itch) [reply src/bmconfigparser.py]: rftd

File: src/class_singleCleaner.py [hunk 19]
```diff
@@ -21,8 +21,8 @@
 
 import gc
 import os
-import shared
 import time
+import shared
```
Comment (g1itch) [inline src/class_singleCleaner.py]:
```python
import time

import knownnodes
import queues
import shared
```

File: src/class_singleCleaner.py [hunk 20]
```diff
@@ -36,11 +36,12 @@
 
 
 class singleCleaner(StoppableThread):
+    """Module run singleCleaner thread"""
```
Comment (g1itch) [inline src/class_singleCleaner.py]: Maybe "The singleCleaner thread class"? What does the module have to do with it?

File: src/class_singleCleaner.py [hunk 21]
```diff
@@ -73,7 +74,7 @@ def run(self):
             # If we are running as a daemon then we are going to fill up the UI
             # queue which will never be handled by a UI. We should clear it to
             # save memory.
-            # FIXME redundant?
+            # ..fixme:: redundant?
```
Comment (g1itch) [inline src/class_singleCleaner.py]: Again, where did you find such reST directive?

File: src/class_smtpServer.py [hunk 22]
```diff
@@ -171,15 +189,19 @@ def run(self):
         asyncore.loop(1)
 
 
-def signals(signal, frame):
+def signals(_, __):
```
Comment (g1itch) [inline src/class_smtpServer.py]: `def signals(*args):` maybe

File: src/class_smtpServer.py [hunk 23]
```diff
@@ -171,15 +189,19 @@ def run(self):
...
+    """Signal handler"""
     logger.warning('Got signal, terminating')
+    print "Got signal, terminating"
```
Comment (g1itch) [inline src/class_smtpServer.py]: remove

File: src/class_smtpServer.py [hunk 24]
```diff
@@ -171,15 +189,19 @@ def run(self):
...
     for thread in threading.enumerate():
         if thread.isAlive() and isinstance(thread, StoppableThread):
             thread.stopThread()
 
 
 def runServer():
+    """Run SMTP server as a standalone python process"""
     logger.warning('Running SMTPd thread')
+    print "Running SMTPd thread"
```
Comment (g1itch) [inline src/class_smtpServer.py]: remove, see #1373

Comment (lakshyacis) [reply src/class_smtpServer.py]: Okay. Thanks

File: src/class_smtpServer.py (hunk 22 above)
Comment (lakshyacis) [reply src/class_smtpServer.py]: Yes, that's looks good. 
Thank you so much.

File: src/class_singleCleaner.py (hunk 21 above)
Comment (lakshyacis) [reply src/class_singleCleaner.py]: I didn't understand what you're trying to say?

File: src/class_singleCleaner.py (hunk 20 above)
Comment (lakshyacis) [reply src/class_singleCleaner.py]: Nothing to do with the module. I write the wrong doc for this class. I will change the docstring and thank you for correcting me.

File: src/class_smtpServer.py (hunk 22 above)
Comment (lakshyacis) [reply src/class_smtpServer.py]: Checks failed ,pylint error for unused variable. **args**

File: src/class_singleCleaner.py (hunk 21 above)
Comment (g1itch) [reply src/class_singleCleaner.py]: What are you trying to achieve by this replacement? Is it for [doc](https://pybitmessage-test.readthedocs.io/en/latest/autodoc/pybitmessage.class_singleCleaner.html)? If so, you should reference the Sphinx [reST syntax](https://www.sphinx-doc.org/en/1.8/usage/restructuredtext/index.html). I cannot found such directive there. It has `..todo:`, but the result [is not very helpful](https://pybitmessage-test.readthedocs.io/en/latest/contribute.dir/develop.dir/todo.html). Also it seems Sphinx [doesn't get them](https://github.com/Bitmessage/PyBitmessage/blob/v0.6/src/network/udp.py#L28) from comments, only from docstrings: https://pybitmessage-test.readthedocs.io/en/latest/autodoc/pybitmessage.network.udp.html.

File: src/bitmessagemain.py [hunk 25]
```diff
@@ -1,4 +1,8 @@
 #!/usr/bin/python2.7
+"""
+src/bitmessagemain.py
+=================================
```
Comment (g1itch) [inline src/bitmessagemain.py]: please remove and write a docstring instead

File: src/class_objectProcessorQueue.py [hunk 26]
```diff
@@ -1,23 +1,32 @@
...
+    """A queue for processsing message objects"""
     maxSize = 32000000
 
     def __init__(self):
         Queue.Queue.__init__(self)
         self.sizeLock = threading.Lock()
-        self.curSize = 0 # in Bytes. We maintain this to prevent nodes from flooing us with objects which take up too much memory. If this gets too big we'll sleep before asking for further objects.
+        # in Bytes. We maintain this to prevent nodes from flooing us with objects
+        # which take up too much memory. If this gets too big we'll sleep before
+        # asking for further objects.
```
Comment (g1itch) [inline src/class_objectProcessorQueue.py]: use `#:`

File: src/class_singleCleaner.py [hunk 27]
```diff
@@ -128,16 +130,16 @@ def run(self):
...
                         os._exit(1)
 
-#            # clear download queues
-#            for thread in threading.enumerate():
-#                if thread.isAlive() and hasattr(thread, 'downloadQueue'):
-#                    thread.downloadQueue.clear()
+            # # clear download queues
+            # for thread in threading.enumerate():
+            #   if thread.isAlive() and hasattr(thread, 'downloadQueue'):
+            #       thread.downloadQueue.clear()
```
Comment (g1itch) [inline src/class_singleCleaner.py]: maybe just remove?

File: src/bitmessagemain.py (hunk 25 above)
Comment (lakshyacis) [reply src/bitmessagemain.py]: Okay. Thank you

File: src/class_singleCleaner.py (hunk 21 above)
Comment (g1itch) [reply src/class_singleCleaner.py]: Look: https://pybitmessage-test.readthedocs.io/en/latest/autodoc/pybitmessage.class_singleCleaner.html
I merged this branch into my `testing`. And there is your hypotetic `.. fixed:` directive result?

File: src/class_objectProcessorQueue.py (hunk 26 above)
Comment (g1itch) [reply src/class_objectProcessorQueue.py]: https://pybitmessage-test.readthedocs.io/en/latest/autodoc/pybitmessage.queues.html#pybitmessage.queues.objectProcessorQueue

File: src/bitmessagemain.py (hunk 25 above)
Comment (lakshyacis) [reply src/bitmessagemain.py]: What should the docstring for bitmessagemain ?

Comment (lakshyacis) [reply src/bitmessagemain.py]: Can you please tell me the docstring for bitmessagemain file.

Comment (g1itch) [reply src/bitmessagemain.py]: """The PyBitmessage startup script""
for example

Comment (g1itch) [reply src/bitmessagemain.py]: Notice, https://github.com/Bitmessage/PyBitmessage/blob/v0.6/docs/conf.py#L227 - this module is excluded from autodoc generation.

Comment (lakshyacis) [reply src/bitmessagemain.py]: Yes, i noticed that. But why it's excluded from autodoc generation?

Comment (g1itch) [reply src/bitmessagemain.py]: because it breaks doc build in current state

Comment (g1itch) [reply src/bitmessagemain.py]: https://pybitmessage-test.readthedocs.io/en/latest/autodoc/pybitmessage.bitmessagemain.html

Comment (lakshyacis) [reply src/bitmessagemain.py]: Okay. Thank you so much

File: src/class_objectProcessorQueue.py (hunk 26 above)
Comment (lakshyacis) [reply src/class_objectProcessorQueue.py]: Can you please elaborate more about this

File: src/class_singleCleaner.py (hunk 21 above)
Comment (lakshyacis) [reply src/class_singleCleaner.py]: > What are you trying to achieve by this replacement? Is it for [doc](https://pybitmessage-test.readthedocs.io/en/latest/autodoc/pybitmessage.class_singleCleaner.html)? If so, you should reference the Sphinx [reST syntax](https://www.sphinx-doc.org/en/1.8/usage/restructuredtext/index.html). I cannot found such directive there. It has `..todo:`, but the result [is not very helpful](https://pybitmessage-test.readthedocs.io/en/latest/contribute.dir/develop.dir/todo.html). Also it seems Sphinx [doesn't get them](https://github.com/Bitmessage/PyBitmessage/blob/v0.6/src/network/udp.py#L28) from comments, only from docstrings: https://pybitmessage-test.readthedocs.io/en/latest/autodoc/pybitmessage.network.udp.html.

Pylint warned me with this FIXME:: **W:135, 0: FIXME redundant? (fixme)**

Comment (lakshyacis) [reply src/class_singleCleaner.py]: > What are you trying to achieve by this replacement? Is it for [doc](https://pybitmessage-test.readthedocs.io/en/latest/autodoc/pybitmessage.class_singleCleaner.html)? If so, you should reference the Sphinx [reST syntax](https://www.sphinx-doc.org/en/1.8/usage/restructuredtext/index.html). I cannot found such directive there. It has `..todo:`, but the result [is not very helpful](https://pybitmessage-test.readthedocs.io/en/latest/contribute.dir/develop.dir/todo.html). Also it seems Sphinx [doesn't get them](https://github.com/Bitmessage/PyBitmessage/blob/v0.6/src/network/udp.py#L28) from comments, only from docstrings: https://pybitmessage-test.readthedocs.io/en/latest/autodoc/pybitmessage.network.udp.html.

Pylint warned me with this FIXME:: **W:135, 0: FIXME redundant? (fixme)**

Comment (lakshyacis) [reply src/class_singleCleaner.py]: > What are you trying to achieve by this replacement? Is it for [doc](https://pybitmessage-test.readthedocs.io/en/latest/autodoc/pybitmessage.class_singleCleaner.html)? If so, you should reference the Sphinx [reST syntax](https://www.sphinx-doc.org/en/1.8/usage/restructuredtext/index.html). I cannot found such directive there. It has `..todo:`, but the result [is not very helpful](https://pybitmessage-test.readthedocs.io/en/latest/contribute.dir/develop.dir/todo.html). Also it seems Sphinx [doesn't get them](https://github.com/Bitmessage/PyBitmessage/blob/v0.6/src/network/udp.py#L28) from comments, only from docstrings: https://pybitmessage-test.readthedocs.io/en/latest/autodoc/pybitmessage.network.udp.html.

Pylint warned me with this FIXME:: **W: FIXME redundant? (fixme)**

Comment (g1itch) [reply src/class_singleCleaner.py]: Just let it warn

File: src/class_objectProcessor.py [hunk 28]
```diff
@@ -765,10 +777,10 @@ def processmsg(self, data):
...
         # messages, disabled or chan
         if (
-            self.ackDataHasAValidHeader(ackData) and not blockMessage and
-            messageEncodingType != 0 and
-            not BMConfigParser().safeGetBoolean(toAddress, 'dontsendack') and
-            not BMConfigParser().safeGetBoolean(toAddress, 'chan')
+                self.ackDataHasAValidHeader(ackData) and not blockMessage and
+                messageEncodingType != 0 and
+                not BMConfigParser().safeGetBoolean(toAddress, 'dontsendack') and
+                not BMConfigParser().safeGetBoolean(toAddress, 'chan')
```
Comment (g1itch) [inline src/class_objectProcessor.py]: Oh, I tired to point you to this pylint bug...

File: src/class_objectProcessorQueue.py (hunk 26 above)
Comment (g1itch) [reply src/class_objectProcessorQueue.py]: It seems I already pointed you to the relevant chapter of the sphinx doc: http://www.sphinx-doc.org/en/1.8/usage/extensions/autodoc.html#directive-autodata

File: src/class_objectProcessor.py (hunk 28 above)
Comment (lakshyacis) [reply src/class_objectProcessor.py]: oh, yes! Thank you . I'll fix!

File: src/class_objectProcessorQueue.py [hunk 29]
```diff
@@ -1,23 +1,31 @@
+"""
+receiveDataThreads dump objects they hear on the network into this queue to be processed
```
[the remark this answers is not in the document]
Comment (lakshyacis) [reply src/class_objectProcessorQueue.py]: I got this docstring from https://pybitmessage-test.readthedocs.io/en/latest/autodoc/pybitmessage.queues.html#pybitmessage.queues.objectProcessorQueue

[the remark this answers is not in the document]
Comment (g1itch) [reply src/class_objectProcessorQueue.py]: I'm sorry. That was an outdated comment which I turned into the annotation mechanically. Fixed in bed2459.

[the remark this answers is not in the document]
Comment (lakshyacis) [reply src/class_objectProcessorQueue.py]: Okay. No problem.
Thank you

File: src/class_objectProcessor.py [hunk 30]
```diff
@@ -1,19 +1,23 @@
+"""
+Processor object threads
```
Comment (g1itch) [inline src/class_objectProcessor.py]: abracadabra

File: src/class_singleCleaner.py (hunk 18 above)
Comment (PeterSurda) [reply src/class_singleCleaner.py]: It looks like this has been resolved

**Comment by PeterSurda** [comment]
@g1itch I have no more objections, you?

**Comment by g1itch** [comment]
> @g1itch I have no more objections, you?

No. It seems working.

**Pull request merged**